### ISSUE TITLE
AWS: Number of Opensearch requests are failing with wrong request signature

### ISSUE THREAD

**Issue by sdorokhova** [camunda/zeebe#17552]
### Describe the bug

When using AWS Opensearch some of he queries fail with wrong request signature error. 
1. When using AWS Opensearch with AWS S3 storage for snapshot repository, Operate (8.5) backup fails with exception:
```
Caused by: io.camunda.operate.exceptions.OperateRuntimeException: Exception occurred when validating whether backup with ID [<..>] already exists.
	at io.camunda.operate.webapp.opensearch.backup.OpensearchBackupRepository.validateNoDuplicateBackupId(OpensearchBackupRepository.java:160) ~[!/:8.5.0]
	at io.camunda.operate.webapp.backup.BackupService.takeBackup(BackupService.java:100) ~[!/:8.5.0]
	at io.camunda.operate.webapp.management.BackupController.takeBackup(BackupController.java:48) ~[!/:8.5.0]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:255) ~[spring-web-6.1.5.jar!/:6.1.5]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:188) ~[spring-web-6.1.5.jar!/:6.1.5]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118) ~[spring-webmvc-6.1.5.jar!/:6.1.5]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:925) ~[spring-webmvc-6.1.5.jar!/:6.1.5]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:830) ~[spring-webmvc-6.1.5.jar!/:6.1.5]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.1.5.jar!/:6.1.5]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.5.jar!/:6.1.5]
	... 120 more
Caused by: org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [security_exception] The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.

```

2. When starting Operate (8.4.6):
```
024-04-18 12:59:04.618 ERROR 8 --- [           main] i.c.o.s.SchemaStartup                    : Schema startup failed: Request failed: [security_exception] The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.

The Canonical String for this request should have been
<..>
	at org.opensearch.client.transport.aws.AwsSdk2Transport.parseResponse(AwsSdk2Transport.java:479) ~[opensearch-java-2.5.0.jar!/:?]
	at org.opensearch.client.transport.aws.AwsSdk2Transport.executeSync(AwsSdk2Transport.java:393) ~[opensearch-java-2.5.0.jar!/:?]
	at org.opensearch.client.transport.aws.AwsSdk2Transport.performRequest(AwsSdk2Transport.java:191) ~[opensearch-java-2.5.0.jar!/:?]
	at io.camunda.operate.opensearch.ExtendedOpenSearchClient.arbitraryRequest(ExtendedOpenSearchClient.java:106) ~[operate-common-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.opensearch.ExtendedOpenSearchClient.arbitraryRequest(ExtendedOpenSearchClient.java:102) ~[operate-common-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.store.opensearch.client.sync.OpenSearchIndexOperations.lambda$getIndexNamesWithRetries$0(OpenSearchIndexOperations.java:44) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.store.opensearch.client.OpenSearchOperation.safe(OpenSearchOperation.java:39) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.store.opensearch.client.sync.OpenSearchIndexOperations.lambda$getIndexNamesWithRetries$2(OpenSearchIndexOperations.java:43) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.store.opensearch.client.sync.OpenSearchSyncOperation.withExtendedOpenSearchClient(OpenSearchSyncOperation.java:26) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.store.opensearch.client.sync.OpenSearchIndexOperations.getIndexNamesWithRetries(OpenSearchIndexOperations.java:42) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.schema.opensearch.OpensearchSchemaManager.getIndexNames(OpensearchSchemaManager.java:116) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.schema.IndexSchemaValidator.hasAnyOperateIndices(IndexSchemaValidator.java:109) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.schema.IndexSchemaValidator.validate(IndexSchemaValidator.java:87) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
	at io.camunda.operate.schema.SchemaStartup.initializeSchema(SchemaStartup.java:44) ~[operate-els-schema-8.4.6.jar!/:8.4.6]
<..>
2024-04-18 12:59:04.630  WARN 8 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.camunda.operate.schema.migration.SchemaMigration': Unsatisfied dependency expressed through field 'schemaStartup': Error creating bean with name 'schemaStartup': Invocation of init method failed
```

### To Reproduce

1. Configure the setup as described here: https://confluence.camunda.com/pages/viewpage.action?pageId=226133974
2. Start Operate of version 8.4.6 or 8.5.0 
3. or call Take backup endpoint on Operate 8.4.6

### Current behavior

Backup fails with the error: `Exception occurred when validating whether backup with ID [<..>] already exists.`

### Expected behavior

Backup is successfully created.

### Environment

- Operate Versions: 8.4.6, 8.5.0
- Opensearch: 2.5
- AWS S3 storage

### Additional context

* Related support issues: https://jira.camunda.com/browse/SUPPORT-20972 and https://jira.camunda.com/browse/SUPPORT-21470
* Same works fine on Tasklist 8.4.6
* Probable problem is in [this part of code](https://github.com/camunda/zeebe/blob/afe4481e720510d108a847026da76de2426cb6b0/operate/common/src/main/java/io/camunda/operate/opensearch/ExtendedOpenSearchClient.java#L138), where we don't propagate request options. Comparing to Tasklist: [here](https://github.com/camunda/tasklist/blob/master/tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/es/backup/os/BackupManagerOpenSearch.java#L364C44-L364C80).
* we have discrepancies in the method `OpenSearchIndexOperations#getIndexNamesWithRetries` in 8.4 and `main` branches, we need to clarify that

---------------------------------------------------------------------------------------------

### Test backup with AWS S3
1. Create a customized docker image for OpenSearch with AWS settings and plugin as this `Dockerfile`:
```
FROM opensearchproject/opensearch:2.5.0
    
ENV OPENSEARCH_INITIAL_ADMIN_PASSWORD StrongPassword1234!
ENV AWS_ACCESS_KEY_ID <INSERT-KEY-ID>
ENV AWS_SECRET_ACCESS_KEY <INSERT-ACCESS-KEY>
    
RUN /usr/share/opensearch/bin/opensearch-plugin install --batch repository-s3
RUN /usr/share/opensearch/bin/opensearch-keystore create
    
RUN echo $AWS_ACCESS_KEY_ID | /usr/share/opensearch/bin/opensearch-keystore add --stdin s3.client.default.access_key
RUN echo $AWS_SECRET_ACCESS_KEY | /usr/share/opensearch/bin/opensearch-keystore add --stdin s3.client.default.secret_key
    
# Optional
RUN echo $AWS_SESSION_TOKEN | /usr/share/opensearch/bin/opensearch-keystore add --stdin s3.client.default.session_token
```

2. Build the docker image and start as OpenSearch container
`docker build -t my-opensearch-s3 .`

3. Register s3 repository
`PUT http://localhost:9200/_snapshot/s3-backups-8.5`
```json
{
  "type": "s3",
  "settings": {
    "bucket": "<INSERT-BUCKET-NAME"
  }
}
```
4. Start Operate with repository name
`camunda.operate.backup.repositoryName=s3-backups-8.5`

5. Now you can test backup like described [here](https://docs.camunda.io/docs/8.4/self-managed/operational-guides/backup-restore/operate-tasklist-backup/) 

### Definition of Ready - Checklist

- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] The issue has been labeled with an appropriate `Bug-area` label
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### :point_right: Handover Dev to QA

- Changed components:
- Side effects on other components:
- Handy resources:
  BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc :
  
- Example projects:

- Commands/Steps needed to test; Versions to validate:

- Docker file / HELM chart : in case that it needed to be tested via docker share the version contain the fixed along with version of other services .

- Release version ( in which version this fixed/feature will be released):

**Comment by ralfpuchert** [comment]
During testing of OpenSearch backup with `s3` repository I found another bug in processing the response from [get snapshot repository endpoint](https://opensearch.org/docs/latest/api-reference/snapshots/get-snapshot-repository/). After fixing this the tests succeded. 
Will prepare another PR for it.

**Comment by ralfpuchert** [comment]
User sees still the same error messages. I checked Tasklist related code parts. One difference is usage of newer OpenSearch Java client library `2.7.0`. Operate uses `2.5.0`.

**Comment by ralfpuchert** [comment]
Point 1. should be solved.
Point 2 is still open.

**Comment by valeriaportolesicamunda** [comment]
[private]
@johanwelgemoed @Zelldon, could you please confirm if it was also released in 8.5.1? I see the label but can't find it in the release note. Thank you!

**Comment by johanwelgemoed** [comment]
[private]
@valeriaportolesicamunda yes, the fix was merged to all supported versions
